src/eval.rs

```diff
@@ -158,9 +161,13 @@ pub fn create_ecx<'mir, 'tcx: 'mir>(
             cur_ptr = cur_ptr.offset(char_size, tcx)?;
         }
     }
- 
+
     assert!(args.next().is_none(), "start lang item has more arguments than expected");
 
+    if config.communicate {
+        EnvVars::init(&mut ecx, &tcx);
+    }
```

Comment:
Please move the `if communicate` logic inside `EnvVars::init` -- the outside world doesn't really have to know such details.

Also, can't you do this immediately after `let mut ecx`? Seems best to initialize the subsystems ASAP.

Reply:
Yes, will do